### DISCUSSION TITLE
After adding Python agent, the service cannot be used normally

### DISCUSSION THREAD

**Discussion by Mushano** [apache/skywalking#10774] [Q&A] [answered]
### Before Adding Python agent
the service can be used normally

### after adding Python agent(whatever 0.8 or 1.0)
any interface related to database queries will report an error like this.

![image](https://user-images.githubusercontent.com/38551745/236755881-0cee605f-afa8-4fa3-b527-02f3916862c4.png)
![image](https://user-images.githubusercontent.com/38551745/236759878-47d0f0d9-4e94-4653-a997-015b29065cde.png)

### View
It seems that the implantation of agent introduces new risks to the service beyond performance.

### Wish
I really want to know Is there a way to avoid the risks posed by agents, such as automatically shutting down agent collection and sending data when detecting errors in the code executed by the agent, so as to ensure the normal operation of the original service.

Sincerely

**Comment by Superskyyy** [comment, marked as answer]
Hello, this is due to an incompatible version of the library is being used with the current Python agent plugins (You can find the tested versions of the library in the agent doc). But indeed what you say is correct, some plugins poses threat to user application while having no fallback mechanism in the case of type conversion like this (while creating spans, it should be try catched), a new issue will be created to set a task to lint current plugins and enforce the fallback mechanism throughout the Python agent. @Mushano

**Reply by Mushano** [reply]
Hi，After careful troubleshooting, it can be determined that the mysqlclient and pymysql libraries are in conflict.

Version:
skywalking-pyton：1.0.1
pymysql:1.0
django:3.2
mysqlclient:2.1.1
![image](https://github.com/apache/skywalking/assets/38551745/f4678f3a-9c46-44d1-bba3-395ff401beeb)


If pymysql and mysqlclient are installed at the same time, an error will be generated when using Django's ORM for database queries.

By looking at the source code of the Python agent, I found that the agent modifies the code through what it call "monkey patching"

And the python agent does not support plugin-in and plugin-out，which can easily cause version conflicts.

I hope that the development of Python Agent can get better and better！

Best wishes！

**Reply by Superskyyy** [reply]
> Hi，After careful troubleshooting, it can be determined that the mysqlclient and pymysql libraries are in conflict.
> 
> 
> 
> Version:
> 
> skywalking-pyton：1.0.1
> 
> pymysql:1.0
> 
> django:3.2
> 
> mysqlclient:2.1.1
> 
> ![image](https://github.com/apache/skywalking/assets/38551745/f4678f3a-9c46-44d1-bba3-395ff401beeb)
> 
> 
> 
> 
> 
> If pymysql and mysqlclient are installed at the same time, an error will be generated when using Django's ORM for database queries.
> 
> 
> 
> By looking at the source code of the Python agent, I found that the agent modifies the code through what it call "monkey patching"
> 
> 
> 
> And the python agent does not support plugin-in and plugin-out，which can easily cause version conflicts.
> 
> 
> 
> I hope that the development of Python Agent can get better and better！
> 
> 
> 
> Best wishes！
> 
> 

I think I heard about these two plugins don't work well together. There's a configuration that you can use to disable plugins, that way you only either use pymysql or mysqlclient.
